src/chatlist.rs

```diff
@@ -91,6 +91,10 @@ impl Chatlist {
         query: Option<&str>,
         query_contact_id: Option<u32>,
     ) -> Result<Self> {
+        // Note that we do not emit DC_EVENT_MSGS_MODIFIED here even if some
+        // messages get hidden to avoid reloading the same chatlist.
+        hide_device_expired_messages(context)?;
```

Comment:
I understand why this is needed.  But I it's an inherently brittle situation I think.

One way out of this might be to never send the msgs-modified even on hiding messages?  Who needs to use this and for what, you have no idea what changed and how it changed.

Another way could be only skip over hidden messages here (and in get_chat_msgs) and hook the actual hiding operation onto something else.  Perhaps on the housekeeping job?  Not sure if that's the best idea ever.

I don't think this should be a blocking thing though, it's just some random brainstorming for ways to avoid this subtle constraint.

Reply:
DC_EVENT_MSGS_MODIFIED is used as a signal for the UIs to refresh chatlist and chats, i think not sending this on hiding/deleting will cause more complexity than this special case where some messages may be removed just before returning the list (in theory,we could do the hide_device_expired_messages() also from a job, however, this would easily result in messages "flashing up and deleted then" - which may be an issue esp. for ppl at risk).

so, i think, the comment does the job just fine, at least for now :)